components/scream/src/physics/shoc/shoc_main_impl.hpp

```diff
@@ -15,6 +15,49 @@ namespace shoc {
  * #include this file, but include shoc_functions.hpp instead.
  */
 
+template<typename S, typename D>
+Int Functions<S,D>::shoc_init(
+  const Int&                  nbot_shoc,
+  const Int&                  ntop_shoc,
+  const view_1d<const Spack>& pref_mid)
+{
+  // This function calculates the maximum number of levels
+  // in pbl from surface
+
+  using ExeSpace = typename KT::ExeSpace;
+  view_1d<Int> npbl_d("npbl",1);
+
+  const auto policy = ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(1, 1);
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const MemberType& team) {
+
+    const Scalar pblmaxp = SC::pblmaxp;
+
+    Int npbl_val = 1;
+
+    const int begin_pack_indx = ntop_shoc/Spack::n;
+    const int end_pack_indx   = nbot_shoc/Spack::n+1;
+    Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, begin_pack_indx, end_pack_indx),
+                                                    [&] (const Int& k, Int& local_max) {
+      auto range = ekat::range<IntSmallPack>(k*Spack::n);
+      const auto condition = (range >= ntop_shoc && range < nbot_shoc && pref_mid(k) >= pblmaxp);
+
+      auto levels_from_surface = nbot_shoc - range;
+      levels_from_surface.set(!condition, 1);
```

Comment:
@ambrad `levels_from_surface` is set to 1 for all indices which do not meet the condition, so that would be the minimum return. I test this case in the bfb test.